### DISCUSSION TITLE
seller/account의 accountId가 반환하는 값이 무엇인지 궁금합니다.

### DISCUSSION THREAD

**Discussion by importunate-dev** [commerce-api-naver/commerce-solutions#93] [묻고 답하기] [answered]
안녕하세요.
https://apicenter.commerce.naver.com/ko/basic/commerce-api#tag/%EC%99%B8%EB%B6%80-%EA%B0%9C%EB%B0%9C%EC%82%AC-%EC%9E%90%EC%B2%B4-%EA%B2%B0%EC%A0%9C/operation/sendTransactionExternal.merchant

여기서 안내하고있는 판매자 API의 https://api.commerce.naver.com/external/v1/seller/account 에 관해 궁금점이 있습니다.
여기서 accountId를 반환하고 있는데, 현재 로그인에 사용하는 계정과 달라 어떤 값이 반환되는것인지 모르겠습니다.
현재 이 스토어가 과거 양도 절차를 통해 양도받은 것으로 알고있는데 혹시 최초 생성시의 ID가 반환되는 것일까요?

**Comment by commerce-api-naver** [comment]
안녕하세요. 커머스API 이민석입니다.

커머스솔루션 연동 과정에서의 `accountUid` 항목은 대상 스토어 계정에 대한 구분자입니다.
이는 판매자 분들께서 로그인에 사용하는 '커머스ID' 계정과는 별개의 값으로 로그인 정보와 관련이 없으며 최초 스토어 생성 시 함께 생성되는 무작위 값입니다.
말씀하신 것과 같이 대상 솔루션을 구독중인 구독자 분께서 과거 스토어 양도양수 절차로 양수받으셨더라도 최초 생성된 스토어의 `accountUid` 항목은 변동되지 않습니다.

**Reply by importunate-dev** [reply]
`accountUid`가 아닌 `accountId`에 관한 질문입니다!

**Reply by commerce-api-naver** [reply]
@junsoopooh 님, 커머스API 이민석입니다.

커머스솔루션 연동에서 구독자 계정 구분은 `accountUid`를 이용하시는 것을 권장하고 있습니다.
혹시 어떠한 사용성이 있어 `accountId` 정보의 사용이 필요하신 상황일까요?

**Reply by importunate-dev** [reply]
민석님 답변 감사합니다.

솔루션을 구독하는 고객의 네이버 ID를 파악할 수 있는지 여부를 확인하는 과정에서 드린 질문이었습니다. 구독자의 고유 값은 accountUid로 사용하고 있지만, 이와 별개로 이 값이 무엇을 의미하는지 궁금하여 질문 드렸습니다.

이 질문과는 별개로 혹시 검수 완료후 사용중인 API 그룹을 추가하고자 할 경우 기존에 구독중인 유저의 액세스토큰을 새로 발급받아야 하는지, 이와 관련된 프로세스가 있는지도 궁금합니다.

**Reply by commerce-api-naver** [reply, marked as answer]
@junsoopooh 님, 커머스API 이민석입니다.

### 구독자 로그인 계정과 `accountUid` 및 `accountId` 관계
먼저 커머스솔루션의 구독자 계정은 <커머스ID>라 불리우는 네이버 커머스 플랫폼 고유의 계정 단위를 사용합니다.
이는 곧 로그인 계정과 동일한 개념입니다.
- 커머스ID = 로그인 계정
    - 이메일 주소로 로그인
    - 네이버 아이디로 로그인

원칙적으로 1개 '커머스ID'는 1개 이상의 스마트스토어 관리 권한을 부여받을 수 있습니다.
이는 '통합매니저'와 같은 스토어의 오너쉽을 갖고 있는 권한일 수도 있으며 매니저 성격의 단순 관리 권한만을 가질 수도 있습니다.
대상하는 커머스ID가 보유한 스마트스토어 관리 권한 중 '통합매니저' 혹은 '주매니저' 권한을 보유한 스토어에 대하여 커머스솔루션 구독이 가능합니다.

따라서 대상 커머스ID가 다수의 스마트스토어의 통합매니저 또는 주매니저 권한을 보유하고 있다면 1개 로그인 계정(커머스ID)으로 여러 스토어에 대해 개별적으로 커머스솔루션 구독이 가능합니다.
그렇기에 커머스솔루션에서 제공하는 개별 사용자의 구분 값은 `accountUid`라 불리는 스토어 단위의 계정 정보를 제공합니다.
- 참고: `accountId` 역시 스토어 단위의 계정 정보이며 이 정보는 장기적 관점에서 사용하지 않는 것을 권장합니다.

### 구독자 커머스ID 정보 미제공
커머스솔루션의 구독 단위는 스마트스토어 단위이기에 구독자의 커머스ID 계정 정보는 커머스API 및 솔루션 연동 토큰(JWT/JWE) 등에서 제공하지 않습니다.
개별 API 역시 구독자의 커머스ID 계정 정보 혹은 대상하는 커머스ID 계정과 연결된 네이버 계정을 활용하지 않는 구조로 설계되었습니다.
- 참고: 판매자의 커머스ID 계정 정보의 전체 혹은 일부를 수집하는 것은 심각한 보안 문제를 일으킬 수 있기에 권장하지 않습니다.

### 추가 API 그룹 사용
솔루션의 심사가 완료된 경우 추가적인 API 그룹 권한은 획득이 불가능합니다.
커머스API센터에서 대상하는 솔루션의 API 그룹 사용 설정은 심사 요청 이전에만 제어가 가능하며, 심사가 완료된 이후 기존 신청한 API 그룹 외 추가적인 API 그룹은 추가할 수 없습니다.
따라서 심사 요청 시 사용하실 API 그룹 권한을 신중히 선택해주시기 바랍니다.

**Reply by importunate-dev** [reply]
친절한 답변 감사합니다. 도움이 많이 되었습니다.
